### DISCUSSION TITLE
Help with 2022 NHL player data from draft_results

### DISCUSSION THREAD

**Discussion by Louque** [mattdodge/yahoofantasy#53] [Q&A] [answered]
Hi I'm new to this project and trying to do as much reading docs and code as I can but am stuck. I'm working my way through some basic python code to do some data analysis on my NHL league. Currently trying to get all drafted players, draft position, and various season (all weeks) stats data.

I'm coming across this issue with the get_stats function while calling it for a player from the draft_results of a league.

![image](https://github.com/mattdodge/yahoofantasy/assets/18126651/346d411d-ab09-41c1-a078-27e8f61b3e77)

I'm wondering if this is an incorrect approach and there's something I'm missing before accessing player data. Or if there's league settings I may need to adjust. I see that game_code member being used in past_league_id(), and mention of a commissioner config.

*Happens with just get_stat(<display name>) as well:

![image](https://github.com/mattdodge/yahoofantasy/assets/18126651/1fcf91d9-fab0-46e8-9d22-b767d6611008)

**Comment by mattdodge** [comment, marked as answer]
It looks like manually creating the league (`League(ctx, id)`) is not storing the game code on there. That's definitely a bug, thanks for pointing it out!

In the meantime, you can solve this a few other ways
1. Add the game code manually (`league.game_code = 'nhl'`)
2. Fetch the league using the `ctx.get_leagues()` method (`league = ctx.get_leagues('nhl', 2023)[0]`)
